client/src/main/java/com/vesoft/nebula/client/graph/net/SyncConnection.java

```diff
@@ -15,23 +15,68 @@
 import com.facebook.thrift.transport.TTransportException;
 import com.facebook.thrift.utils.StandardCharsets;
 import com.vesoft.nebula.ErrorCode;
+import com.vesoft.nebula.client.graph.data.CASignedSSLParam;
 import com.vesoft.nebula.client.graph.data.HostAddress;
+import com.vesoft.nebula.client.graph.data.SSLParam;
+import com.vesoft.nebula.client.graph.data.SelfSignedSSLParam;
 import com.vesoft.nebula.client.graph.exception.AuthFailedException;
 import com.vesoft.nebula.client.graph.exception.IOErrorException;
 import com.vesoft.nebula.graph.AuthResponse;
 import com.vesoft.nebula.graph.ExecutionResponse;
 import com.vesoft.nebula.graph.GraphService;
+import com.vesoft.nebula.util.SslUtil;
+import java.io.IOException;
+import javax.net.ssl.SSLSocketFactory;
 
 public class SyncConnection extends Connection {
     protected TTransport transport = null;
     protected TProtocol protocol = null;
     private GraphService.Client client = null;
     private int timeout = 0;
+    private SSLParam sslParam = null;
+    private boolean enabledSsl = false;
+
+    @Override
+    public void open(HostAddress address, int timeout, SSLParam sslParam) throws IOErrorException {
+        try {
+            SSLSocketFactory sslSocketFactory;
+
+            this.sslParam = sslParam;
+            if (sslParam == null) {
```

Comment:
why need this? 
if sslParm is null, ConnObjectPool.java would raise IllegalArgumentException